docs/plugins/events.md

```diff
@@ -1,45 +1,138 @@
 # Events
 
-Generic Events emitter / listener interface which can be used for _any_ type of event. For Lifecycle events please see [Lifeycle](/plugins/lifecycle.md)
+_work in progres_
+
+Events plugin is used to listen to events emitted from:
+
+- various SDK plugins (i.e. VideoPlayer plugin, Router plugin)
+- the transport layer (for realtime changes in the Platform or Lifecycle)
+- custom App events broadcasted from the App code
 
 ## Usage
 
-The SDK comes with a generic Eventing mechanism which can be, optionally, used within the application. To start:
+In order to use the Events plugin, import it from the Lightning SDK.
 
 ```js
 import { Events } from '@lightningjs/sdk'
 ```
 
 ## Available methods
 
-### Listening for events
+### Listen
+
+Listen to events and execute a callback when they occur.
+
+Examples:
+
+1) Listen to a specific event on a specific Plugin
+
+```js
+Events.listen('Lifecycle', 'pause', () => {
+  console.log('The App is going in a pause state')
+})
+
+Events.listen('Platform', 'device.hdcp', val => {
+  console.log('The hdcp value changed on the platform!', val)
+})
+```
+
+2) Listen to all events of a specific Plugin
+
+```js
+Events.listen('Lifecycle', (val, plugin, event) => {
+  if(event === 'init') {
+    // execute init functionality
+  }
+  if(event === 'close') {
+    // execute close functionality
+  }
+})
+```
+
+3) Listen to all events
+
+```js
+Events.listen((val, plugin, event) => {
+  if(plugin === 'Router' && event === 'pageLoad') {
+    // a new page was loaded
+  }
+  if(plugin === 'VideoPlayer' && event === 'play') {
+    // video player started playing
+  }
+})
+```
 
-To listen for events you can set a `addEventListener` providing a `event` and a `callback` function to be called when the event happens
+Note that it's possible to register multiple listeners for the same event
 
 ```js
-Events.addEventListener('mycoolevent', myCoolFunction)
+Events.listen('Platform', 'device.hdcp', val => {
+  console.log('Listen once!', val)
+})
+
+Events.listen('Platform', 'device.hdcp', val => {
+  console.log('Listen twice!', val)
+})
 ```
 
-### Remove event listener
+### Clear
+
+Registered events can be cleared
+
+1) By id
+
+```js
+const listenerId = Events.listen('Platform', 'device.dhcp', () => {})
```

Comment:
Providing an empty callback to indicate "I'm not listening anymore!" is odd to me. Is there any reason not to coat this with a little sugar like `Events.clear('Platform', 'device.hdcp')`?

Reply:
Oh sorry, I should have scrolled down some more :)

Reply:
Yup, this was just a quick and dirty example. The callback in the listener is just a void function. And it's clearing it by id.
Agreed that in the final docs that should be a more clear example.